syllabus/core/curriculum_base.py

```diff
@@ -150,13 +151,25 @@ def _sample_distribution(self) -> List[float]:
         raise NotImplementedError
 
     def _should_use_startup_sampling(self) -> bool:
-        return self.random_start_tasks > 0 and self.completed_tasks < self.random_start_tasks
-
-    def _startup_sample(self) -> List:
-        task_dist = [0.0 / self.num_tasks for _ in range(self.num_tasks)]
-        task_dist[0] = 1.0
-        return task_dist
-
+        return self.use_startup_behavior 
```

Comment:
```suggestion
        return self.warmup_strategy != "none" and self.sampled_tasks < self.warmup_tasks
```

Your previous implementation seemed closer to what I'm looking for.